app/controllers/application_controller/ci_processing.rb

```diff
@@ -872,10 +872,13 @@ def discover
             end
             Host.discoverByIpRange(from_ip, to_ip, options)
           else
-            if params[:discover_type_selected] == ExtManagementSystem.ems_cloud_discovery_types['azure']
-              ManageIQ::Providers::Azure::CloudManager.discover_queue(@client_id, @client_key, @azure_tenant_id)
+            cloud_manager = ManageIQ::Providers::CloudManager.subclasses.find do |ems|
```

Comment:
Prefer `detect` over `find` (this awkwardly isn't in the bbatsov guide, but our [own fork](https://github.com/ManageIQ/ruby-style-guide))